src/test/groovy/graphql/normalized/NormalizedQueryFactoryTest.groovy

```diff
@@ -1103,7 +1103,93 @@ schema {
         arg2.value == expectedNormalizedArgValue
         arg3.getTypeName() == "Input1"
         arg3.value == expectedNormalizedArgValue
+    }
+
+    def "normalized arguments with absent variable values inside input objects"() {
+        given:
+        def schema = """
+        type Query {
+            hello(arg: Arg, otherArg: String): String
+        }
+        input Arg {
+            ids: [ID]
+        }
+        """
+        def graphQLSchema = TestUtil.schema(schema)
+
+        def query = """
+        query myQuery(\$varIds: [ID], \$otherVar: String) {
+            hello(arg: {ids: \$varIds}, otherArg: \$otherVar)
+        }
+        """
+
+        assertValidQuery(graphQLSchema, query)
+        def document = TestUtil.parseQuery(query)
+        def dependencyGraph = new NormalizedQueryFactory()
+        def variables = [:]
+        when:
+        def tree = dependencyGraph.createNormalizedQueryWithRawVariables(graphQLSchema, document, null, variables)
 
+        then:
+        def topLevelField = tree.getTopLevelFields().get(0)
+        def arg = topLevelField.getNormalizedArgument("arg")
+        def otherArg = topLevelField.getNormalizedArgument("otherArg")
+
+        arg == new NormalizedInputValue(
+                "Arg",
+                [
+                        ids: new NormalizedInputValue(
```

Comment:
they should not be `null`: a missing var value is different from a `null` one.  It should be an empty map.